native/packages/react-native-bpk-component-horizontal-nav/src/BpkHorizontalNav.js

```diff
@@ -35,48 +40,157 @@ const styles = StyleSheet.create({
+    const { width, x } = event.nativeEvent.layout;
+    this.childrenPositions[index] = { width, x };
+
+    // If the child in question is the initially selected one,
+    // the indicator can now be positioned.
+    if (this.props.selected === index) {
+      this.setState({
+        indicatorOffsetX: new Animated.Value(x),
+        indicatorWidth: new Animated.Value(width),
+      });
+    }
+  }
+  animateIndicator = ({ x, width }) => {
+    const duration = 200;
+    Animated.parallel([
+      Animated.timing(this.state.indicatorOffsetX, {
+        toValue: x,
+        duration,
+      }),
+      Animated.timing(this.state.indicatorWidth, {
+        toValue: width,
+        duration,
+      }),
+    ]).start();
+  }
+  childrenPositions = {};
+  render() {
+    const {
+      children,
+      selected,
+      spaceAround,
+      style,
+      theme,
+      ...rest
+    } = this.props;
+
+    if (selected && !validateSelectedIndex(selected, React.Children.count(children))) {
+      throw new Error(`Invalid \`selected\` prop. Must be an integer between 0 and ${React.Children.count(children) - 1}.`); // eslint-disable-line max-len
+    }
+
+    const navStyle = [styles.horizontalNav];
+    if (spaceAround) {
+      navStyle.push(styles.spaceAround);
+    }
+
+    const enhancedChildren = (selected !== null) ?
+      React.Children.map(children, (child, index) => (
+        React.isValidElement(child) && React.cloneElement(child, {
+          key: index, // eslint-disable-line react/no-array-index-key
+          selected: child.props.selected || selected === index,
+
+          // Have children report their layout details after being laid out.
+          // This allows the selected indicator to be correctly positioned.
+          onLayout: event => this.onChildLayout(event, index),
+        }))) : children;
+
+    const renderSelectedIndicator = (selectedIndex) => {
+      if (this.state.indicatorOffsetX && selectedIndex !== null) {
+        const indicatorStyle = [styles.selectedIndicator];
+        if (theme && theme[THEMING_ATTRIBUTE]) {
+          indicatorStyle.push({ backgroundColor: theme[THEMING_ATTRIBUTE] });
+        }
+        const animatedStyles = {
+          transform: [{
+            translateX: this.state.indicatorOffsetX,
+          }],
+          width: this.state.indicatorWidth,
+        };
+        return <Animated.View style={[indicatorStyle, animatedStyles]} />;
+      }
+      return null;
+    };
+
+    return (
+      <ScrollView
+        alwaysBounceHorizontal={false}
+        contentContainerStyle={[navStyle, style]}
+        horizontal
+        showsHorizontalScrollIndicator={false}
+        {...rest}
+      >
+        <View>
+          <View style={styles.horizontalNavInner}>
+            {enhancedChildren}
+          </View>
+          {renderSelectedIndicator(selected)}
+        </View>
+      </ScrollView>
+    );
+  }
+}
+
+const propTypes = {
+  children: PropTypes.arrayOf(PropTypes.element).isRequired,
```

Comment:
Doesn't `PropTypes.element` already support one or more?

Reply:
When I tried it, I got proptypes errors from having more than one. Will double-check though.

Reply:
Fair, I might be thinking of node